_docs/src/documentation/rex_pipeline.md

```diff
@@ -105,40 +105,70 @@ It'll scan for all sharpmake files recursively in 3 different folders
 
 all these files are then passed on to sharpmake and executed.
 
-The generation script also has some config flags to enable or disable some behavior done in the generation step.
+Because the generation script is a wrapper around a sharpmake call, the script itself doesn't own the commandline arguments, these are owned by Sharpmake.
+We needed a way to let sharpmake communicate with the python script and back so that we don't have to maintain the same list of commandlines twice, once in sharpmake and once in the python script. We've achieved this by creating a `default_config.json` file that lives next to the sharpmake build sources
 
-| argument                  |  description 
-|--                         |--:            
-| -no_clang_tools           | disables clang tools in any post build command  
-| -unit_tests               |  also generates unit test projects 
-| -coverage                 |  adds code coverage flags to compilation steps 
-| -asan                     | adds code address sanitizer to compilation steps  
-| -ubsan                    | adds code undefined behavior sanitizer to compilation steps  
-| -fuzzy                    |  adds code fuzzy testing to compilation steps 
-| -sharpmake_args           | pass in additional arguments to sharpmake  
+This config file is parsed by the python script and it'll create commandline argument for them. the script will create a `config.json` file in the intermediate build folder which is passed to sharpmake. It'll read the config file and use it to initialize the generation pipeline.
+
+There's only 1 extra commandline argument supported by the generate script which is `-sharpmake_args`. This is to allow a user to pass in arguments that are supported by the native sharpmake executable
+
+We support Visual Studio and Visual Studio Code as IDEs. These just act as text editors as they're in turn call into the build pipeline and build the ninja files which are always generated.
```

Comment:
Ah oke, so nothing is changed on disk if it's the same thing, then it's oke. So in a way they are cached.
I would just opt to reuse the ninja file if it's the same thing as before.

Reply:
it can't know if it's the same things as before until it has generated it. there could be new files, new configuration settings, sharpmake could've been updated, ... there are many variables at play, only after the generation has happened it can compare its final result with what's on disk, if there's no difference it won't do a file write operation and therefore not update the modification date. This has in turn the effect that no full rebuild will be performed as the file hasn't changed

Reply:
Well you could know it but that would require a lot more work though. so it's fine.